conans/model/conan_file.py

```diff
@@ -351,3 +351,23 @@ def set_deploy_folder(self, deploy_folder):
         self.buildenv_info.deploy_base_folder(self.package_folder, deploy_folder)
         self.runenv_info.deploy_base_folder(self.package_folder, deploy_folder)
         self.folders.set_base_package(deploy_folder)
+
+
+@contextmanager
+def use_build_context(conanfile):
```

Comment:
I think I prefer this explictly in the MesonToolchain, like: ``settings = self.settings_build if not native else self.settings``.

Reply:
Yeah, the thing is that we would have to change all the other `*_build` variables (options, conf, even buildenv and runev?)